packages/extension-api/src/extension-api.d.ts

```diff
@@ -788,6 +789,131 @@ declare module '@podman-desktop/api' {
     connection: ContainerProviderConnection;
   }
 
+  /**
+   * Callback for openning shell session
+   */
+  export interface ProviderConnectionShellAccess {
+    /**
+     * Opens new session using ProviderConnectionShellAccessImpl class
+     * @example
+     * ```typescript
+     * open(): ProviderConnectionShellAccessSession {
+     *  // Create stream 
+     *  #connected = true;
+     *  return {
+     *    onData: onData,
+     *    onError: onError,
+     *    onEnd: onEnd,
+     *    write: write.bind(this),
+     *    resize: resize.bind(this),
+     *    close: close,
```

Comment:
when left and right part is the same, you can just use the name

return { onData}

but still, if I look at this example, how will it help me ? it seems here I won't do anything with this example